tests/test_parser.py

```diff
@@ -390,6 +398,38 @@ def test_if_else_expr_expr():
     check_ast("42+5 if 1 == 2 else 65-5")
 
 
+def test_subscription_syntaxes():
+    assert eval_code("[1, 2, 3][-1]") == 3
+    assert eval_code("[1, 2, 3][-1]") == 3
+    assert eval_code("'string'[-1]") == "g"
+
+
+@pytest.fixture
+def arr_container():
+    # like numpy.r_
+    class Arr:
+        def __getitem__(self, item):
+            return item
+
+    return Arr()
+
+
+def test_subscription_special_syntaxes(arr_container):
+    assert eval_code("arr[1, 2, 3]", arr=arr_container) == (1, 2, 3)
+    # dataframe
+    assert eval_code('arr[["a", "b"]]', arr=arr_container) == ["a", "b"]
+
+
+# todo: enable this test
+# def test_subscription_special_syntaxes_2(arr_container):
+#     # aliases
+#     d = {}
+#     eval_code("d[arr.__name__]=True", arr=arr_container, d=d)
+#     assert d == {"Arr": True}
+#     # extslice
+#     assert eval_code('arr[:, "2"]') == 2
```

Comment:
Maybe mark this as an xfail instead of commenting out?

Reply:
I wanted to create another PR where one could work on making this test pass. I think it is better to mark it instead of commenting. I will update it.